lib/eth/advance/helper.ex

```diff
@@ -0,0 +1,170 @@
+
+  Returns `{signature, r, s, v}`
+  """
+  @spec get_signature(binary(), String.t()) :: {binary(), integer(), integer(), integer()}
+  def get_signature(transaction_digest, private_key) do
+    Signature.sign_digest(
+      transaction_digest,
+      Utils.hex_to_binary(private_key)
+    )
+  end
+
+  @doc """
+  Gets the digest of the transaction to sign.
+  """
+  @spec get_transaction_to_sign(
+          integer(),
+          integer(),
+          integer(),
+          String.t(),
+          integer(),
+          String.t()
+        ) :: binary()
+  def get_transaction_to_sign(nonce, gas_price, gas_limit, to, value, input) do
+    # EIP155 spec:
+    # when computing the hash of a transaction for purposes of signing or recovering,
+    # instead of hashing only the first six elements (ie. nonce, gasprice, startgas, to, value, data),
+    # hash nine elements, with v replaced by CHAIN_ID, r = 0 and s = 0
+    chain_id = @chain_ids[:ethereum_mainnet]
+
+    nonce
+    |> new_transaction(gas_price, gas_limit, to, value, input, chain_id, 0, 0)
+    |> ExRLP.encode()
+    |> Keccak.kec()
+  end
+
+  @doc """
+  Gets the raw transaction which is ready to broadcast to the network.
+  """
+  @spec get_raw_transaction(
+          integer(),
+          integer(),
+          integer(),
+          String.t(),
+          integer(),
+          String.t(),
+          integer(),
+          integer(),
+          integer()
+        ) :: String.t()
+  def get_raw_transaction(nonce, gas_price, gas_limit, to, value, input, v, r, s) do
+    recovery_id = get_recovery_id(v)
+
+    nonce
+    |> new_transaction(gas_price, gas_limit, to, value, input, recovery_id, r, s)
+    |> ExRLP.encode()
+    |> Base.encode16(case: :lower)
+  end
+
+  @doc """
+  Construct a transaction object in the Ethereum format, which is a list of nince elements' binary values:
+  1. `nonce`: The sender's nonce at the current block. Evaluated as empty binary array (<<>>) if it is zero.
+  2. `gas_price`:
+  3. `gas_limit`:
+  4. `to`: Receiver's address. Evaluated as empty binary array (<<>>) if it is null.
+  5. `value`: The amount of Wei to transact. Evaluated as empty binary array (<<>>) if it is null or zero.
+  6. `input`: The input data. Evaluated as empty binary array (<<>>) if it is null.
+  7. `v`: Recover Id. Use chain_id to get a transaction data to sign; use either 37 or 38 (EIP-155) to get the raw transaction to broadcast.
+  8. `r`: r value of the ECDS signature. Evaluated as empty binary array (<<>>) if it is null or zero.
+  9. `v`: v value of the ECDS signature. Evaluated as empty binary array (<<>>) if it is null or zero.
+  """
+  @spec new_transaction(
+          integer(),
+          integer(),
+          integer(),
+          String.t(),
+          integer(),
+          String.t(),
+          integer(),
+          integer(),
+          integer()
+        ) :: list(binary())
+  def new_transaction(nonce, gas_price, gas_limit, to, value, input, v, r, s) do
```

Comment:
better to be: `def new_transaction(to, value, opts \\ [])` and opts here is a keyword list. This will make the interface easy and if the given value is not there the function will put a value with the best effort.